File: backend/app/schemas/closing.py
# app/schemas/closing.py
# -*- coding: utf-8 -*-
from typing import List, Dict, Literal, Optional
from pydantic import BaseModel, Field, ConfigDict


# Calendar/day rows
class ClosingDayRow(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    date: str
    uploaded: List[str] = Field(default_factory=list)
    counts: Dict[str, int] = Field(default_factory=dict)
    done: int = 0
    total: int = 0
    complete: bool = False
    status: Literal["OPEN", "CLOSED"] = "OPEN"


class ClosingCalendarResp(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    property_code: str
    timezone: str = "UTC"
    month: Optional[str] = None
    from_: str = Field(alias="from")
    to: str
    required: List[str] = Field(default_factory=list)
    days: List[ClosingDayRow] = Field(default_factory=list)


# Day get/set
class DayStatusBody(BaseModel):
    date: str
    property_code: str = "MOP"
    status: Literal["OPEN", "CLOSED"]


# Restore
class RestoreBody(BaseModel):
    dataset: str
    business_date: str
    property_code: str = "MOP"
    version_no: int

File: frontend/admin/src/services/closing.ts
import http from '@/services/http'

/** ─────────────────────────────
 * 1) 일자 단위 마감 상태 조회/변경
 *  (기존 /closing/day 유지)
 * ───────────────────────────── */
export async function getClosingDay(params: { date: string; property_code: string }) {
  return await http.get(`/closing/day${http.qs(params)}`)
}

export async function setClosingDayStatus(body: { date: string; status: 'OPEN'|'CLOSED'; property_code: string }) {
  // ✅ 백엔드가 JSON Body 또는 Query를 받도록 되어있음.
  //   기존 FormData → JSON으로 변경
  return await http.put('closing/day', body)
}

/** ─────────────────────────────
 * 2) 월 단위 캘린더 조회 (/api/closing/calendar)
 *  BE가 from/to(days[] 범위) 계산 — FE는 그대로 표시만
 * ───────────────────────────── */
export type ClosingDay = {
  date: string
  uploaded: string[]
  counts: Record<string, number>
  done: number
  total: number
  complete: boolean
  status: 'OPEN' | 'CLOSED'
}

export type ClosingCalendarResp = {
  ok: boolean
  property_code: string
  month: string
  timezone: string
  from: string
  to: string
  required: string[]
  days: ClosingDay[]
}

export async function getClosingCalendar(params: { month: string; property_code: string }) {
  return await http.get<ClosingCalendarResp>(`/closing/calendar${http.qs(params)}`)
}
